src/Optimizer/Transformer/MinifyHtml.php

```diff
@@ -253,21 +262,64 @@ private function minifyJson(DOMText $node, ErrorCollection $errors)
         $decodedData = json_decode($node->data);
 
         if (JSON_ERROR_NONE !== json_last_error()) {
-            $errors->add(InvalidJson::fromLastErrorMsgAfterDecoding());
+            $errors->add(Error\InvalidJson::fromLastErrorMsgAfterDecoding());
         }
 
         if (! empty($decodedData)) {
             $data = json_encode($decodedData, JSON_HEX_AMP | JSON_HEX_TAG | JSON_UNESCAPED_SLASHES);
 
             if (JSON_ERROR_NONE !== json_last_error()) {
-                $errors->add(InvalidJson::fromLastErrorMsgAfterEncoding());
+                $errors->add(Error\InvalidJson::fromLastErrorMsgAfterEncoding());
             }
 
             // PHP uses uppercase letters for angle bracket codes, so convert them into lowercase.
             $node->data = str_replace(['\u003E', '\u003C'], ['\u003e', '\u003c'], $data);
         }
     }
 
+    /**
+     * Checks if a node is meant to be an inline amp-script.
+     *
+     * @param Element $node The element node to be checked.
+     * @return bool
+     */
+    private function isInlineAmpScript(Element $node)
+    {
+        $type = $node->getAttribute(Attribute::TYPE);
+        $target = $node->getAttribute(Attribute::TARGET);
+
+        return $type === Attribute::TYPE_TEXT_PLAIN || $target === Protocol::AMP_SCRIPT;
+    }
+
+    /**
+     * Minify inline AMP script.
+     *
+     * @param DOMText         $node   The node to be minified.
+     * @param ErrorCollection $errors Collection of errors that are collected during transformation.
+     */
+    private function minifyAmpScript(DOMText $node, ErrorCollection $errors)
+    {
+        if (! class_exists(Peast::class) || ! class_exists(Renderer::class) || ! class_exists(Compact::class)) {
+            $errors->add(Error\MissingPackage::withMessage(
+                'The optional package mck89/peast package is required to minify inline amp-script.'
+            ));
+            return;
+        }
+
+        // @codeCoverageIgnoreStart
```

Comment:
Why is code coverage being ignored here? You should probably be able to trigger an exception from Peast in some way, no?

Reply:
@schlessera I can test this in my local by manually adding the Peast, but how can I handle this for GitHub automated testing? I cannot use `createMock` either, cause Peast is not used as the constructor param. Should I create dummy classes just for testing purpose while testing this piece of code? Please suggest. Thanks.

Reply:
Ah, right. Leave it as is for now, then, and create an issue noting that Peast is currently not being test and that testing it will require a change to the GHA testing setup.

Reply:
Created a new issue for the GitHub action https://github.com/ampproject/amp-toolbox-php/issues/315